server/src/engineModel/EngineModel.ts

```diff
@@ -255,7 +257,33 @@ export class EngineModel {
      */
     private enrichYmlClass(yclass: YmlClass): void {
         yclass.data = `id_${yclass.label}`;
-        yclass.detail =
-            yclass.extends.length > 0 ? `Class: ${yclass.label} extends ${yclass.extends}` : `Class: ${yclass.label}`;
+        yclass.detail = `(class) ${yclass.label}`;
+        if (yclass.extends.length > 0) {
+            yclass.detail = yclass.detail.concat(` extends ${yclass.extends}`);
+        }
     }
 }
+
+/**
+ * Returns the value of first xmlElement's child named `doc` tag if any.
+ *
+ * E.g. for the following XML:
+ * ```xml
+ * <xmlElement>
+ *     <doc>documentation</doc>
+ * </xmlElement>
+ * ```
+ * its representation is:
+ * ```JSON
+ * {
+ *     doc: ["documentation"]
```

Comment:
* Not sure what you mean by “representation”. Is it what the function returns?
* The value seems to end up in a list, here (`[` `]`), but your doc says nothing of the sort (and the code seems to fetch `[0]`).

Reply:
* We're using a package (`xml2js`) which transforms XML into a JSON representation. 
* Basically, this package creates a new object structure in which each XML child is an attribute named as the XML node and which value is an array. This array will contain the content of each node sharing the same name. The value can be an object or a string.

I'm not sure I'm very clear. Sorry.

Reply:
Should the caller of the function give a damn about that internal JSON representation? Oh, you mean that we must pass as an argument a piece of XML that follows this representation. That could be explained in a better way, yes, I think xD

Reply:
Isn't the function documentation clear enough?
> Returns the value of the first child `doc` element within a given XML element, if any.

Reply:
No, the documentation is misleading at best. You say “e.g.”, so I expect an example of input→output, and what you give instead are weird guidelines for how XML should be represented before calling the function.

I’d

1) replace “its representation is” with “it must be given as the following JSON representation, obtained thanks to xml2js”

2) add the corresponding function output, otherwise it’s not an example (or at least not one logically linked to the first sentence of the docs). Then the “e.g.” will make more sense.

Reply:
:+1:

Reply:
Something like that?
![image](https://user-images.githubusercontent.com/15327919/81165349-1e071800-8f92-11ea-8091-4ae201bc2fcb.png)

Reply:
Aye.

Reply:
Done.